src/broadcast/broadcast.rs

```diff
@@ -218,9 +243,13 @@ impl<N: NodeIdT> Broadcast<N> {
         if !self.validate_proof(&p, &self.our_id()) {
             return Ok(Fault::new(sender_id.clone(), FaultKind::InvalidProof).into());
         }
-
-        // Otherwise multicast the proof in an `Echo` message, and handle it ourselves.
-        self.send_echo(p)
+        // Store value received to send back at later stage.
+        self.value_message = p.clone();
+        // Multicast the proof in an `Echo` message to left nodes
```

Comment:
Let's reserve the word "multicast" for instances where messages are actually sent to _all_ other validators.

Reply:
Wouldn't sending to all other validators be broadcast?!

Reply:
I think the Honey Badger BFT paper uses the term "multicast" to avoid confusion with the "Reliable Broadcast Protocol", and we adopted that. Since we call the latter just "Broadcast", we should be even more careful.